openfisca_core/formulas.py

```diff
@@ -433,162 +481,61 @@ def compute(self, period, **parameters):
                     simulation.stringify_input_variables_infos(input_variables_infos), str(period),
                     stringify_array(array)))
 
-        dated_holder = holder.put_in_cache(array, period, extra_params)
-
         self.clean_cycle_detection_data()
         if max_nb_cycles is not None:
             simulation.max_nb_cycles = None
 
-        return dated_holder
+        return holders.DatedHolder(self.holder, period, array, extra_params)
+
+    def find_function(self, period):
+        """
+        This function finds the last active formula for the time interval [period starting date, variable end attribute].
```

Comment:
Done (it was used to act like the python official documentation but it's not even consistent there).